sequence/src/main/java/com/vitreoussoftware/bioinformatics/sequence/encoding/BasicDnaEncodingScheme.java

```diff
@@ -61,7 +63,19 @@ public static BasePair create(char nucleotide) throws InvalidDnaFormatException
 	public BasePair fromCharacter(Character character) {
 		return BasicDnaEncodingScheme.create(character);
 	}
-	
+
+	@Override
+	public BasePair complement(BasePair basePair) {
+		if (!basePair.getEncodingScheme().equals(this)) {
+			return complement(fromCharacter(basePair.toChar()));
+		}
+
+		byte nucleotide = basePair.getValue();
+		if ((nucleotide & SHIFT_RIGHT) != 0)
+			return toBasePair((byte) (nucleotide >>> 1));
+        return toBasePair((byte) (nucleotide << 1));
```

Comment:
Might be clearer if you use an if / else clause in terms of legibility.  Makes it clear you're returning at this point using one of two methods.